src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -677,9 +683,9 @@ public bool Remove(TKey key)
                 Debug.Assert(_entries.Length > 0, "entries should be non-empty");
                 uint collisionCount = 0;
 
-                IEqualityComparer<TKey>? comparer = _comparer;
-                Debug.Assert(typeof(TKey).IsValueType || comparer is not null);
-                var hashCode = (uint)(typeof(TKey).IsValueType && comparer == null ? key.GetHashCode() : comparer!.GetHashCode(key));
+                var comparer = _comparer;
+                Debug.Assert((SupportsComparerDevirtualization && typeof(TKey).IsValueType) || comparer is not null);
```

Comment:
Similar comment - unclear why this is needed for Roslyn build.

Reply:
➡️ It appears the change to `var` here was not required. IDE0007 was showing up in the IDE (suggesting the use of `var`), but it could have been overlooked since the severity was lower than warning. Since the result of the change is consistent through the file and the merge history is particularly complex, I'm planning to leave this one as is.

---

➡️ `SupportsComparerDevirtualization` is unique to our build. The .NET Runtime reference implementation applied some optimizations to `Dictionary<TKey, TValue>` based on changes to the JIT that improved the code generated for `EqualityComparer<T>.Default.Equals` and `EqualityComparer<T>.Default.GetHashCode`. These optimizations result in certain cases where the `_comparer` field is `null` (this is the trigger for using `EqualityComparer<TKey>.Default` directly), which has a side effect that we can't blindly invoke `_comparer.GetHashCode` here. When `SegmentedDictionary<TKey, TValue>` was created and retained support for .NET Framework, `SupportsComparerDevirtualization` was added as a way to separate the optimized code paths for .NET Framework and .NET execution scenarios. During this update, the reference code in .NET slightly altered the conditions under which it uses `EqualityComparer<TKey>.Default` directly, so I merged the changes we made for .NET Framework with the changes made for .NET.

The easiest way to understand these changes is the reference code for .NET 5 was unconditional (implicit `true`), while .NET 8 is conditional on `typeof(TKey).IsValueType`. This means the condition in our modified code went from `SupportsComparerDevirtualization && true` to `SupportsComparerDevirtualization && typeof(TKey).IsValueType`.